includes/pattern.inc

```diff
@@ -134,30 +134,53 @@ function oe_bootstrap_theme_preprocess_pattern_navbar(array &$variables): void {
  * Implements hook_preprocess_HOOK() for featured media pattern.
  */
 function oe_bootstrap_theme_preprocess_pattern_featured_media(&$variables) {
-  if (!isset($variables['image'])) {
-    return;
+  if (isset($variables['image'])) {
+    $variables['image'] = oe_bootstrap_theme_prepare_pattern_image($variables['image']);
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK() for inpage navigation pattern.
+ */
+function oe_bootstrap_theme_preprocess_pattern_inpage_navigation(&$variables): void {
+  $variables['inpage_navigation_id'] = $variables['attributes']['id'] ?? Html::getUniqueId('bcl-inpage-navigation');
+}
+
+/**
+ * Implements hook_preprocess_HOOK() for content banner.
+ */
+function oe_bootstrap_theme_preprocess_pattern_content_banner(array &$variables) {
+  if (isset($variables['image'])) {
+    $variables['image'] = oe_bootstrap_theme_prepare_pattern_image($variables['image']);
   }
+}
 
+/**
+ * Prepare the image to render.
+ *
+ * @param array $image
+ *   An array with the ImageValueObject params.
+ *
+ * @return array
+ *   An image render array.
+ */
+function oe_bootstrap_theme_prepare_pattern_image(array $image): array {
+  if (empty($image)) {
+    return [];
+  }
   // This conversion is needed only to run markup rendering tests, as they
   // are reading from a yaml file where we cannot specify objects.
   // Do not ever rely on this for your patterns.
   // @todo Remove this! Code duplication below is intentional, these lines are
   //   to be removed, do not refactor unless to delete them.
-  if (is_array($variables['image']) && !empty($variables['image'])) {
-    $variables['image'] = ImageValueObject::fromArray($variables['image']);
-  }
+  $image = ImageValueObject::fromArray($image);
 
-  if ($variables['image'] instanceof ImageValueObjectInterface) {
+  if ($image instanceof ImageValueObjectInterface) {
     // Transform an object implementing ImageValueObjectInterface to our own
     // implementation, so that we can use the specific method.
-    $variables['image'] = ImageValueObject::fromArray($variables['image']->getArray())
+    $image = ImageValueObject::fromArray($image->getArray())
       ->toRenderArray();
   }
-}
 
-/**
- * Implements hook_preprocess_HOOK() for inpage navigation pattern.
- */
-function oe_bootstrap_theme_preprocess_pattern_inpage_navigation(&$variables): void {
-  $variables['inpage_navigation_id'] = $variables['attributes']['id'] ?? Html::getUniqueId('bcl-inpage-navigation');
+  return $image;
```

Comment:
Please revert all of the changes to other functions.
Then let's focus on what is supposed to be passed to the content banner: an ImageValueObject instance.
What is needed in the BCL template? Path and alt. So there's no need for any preprocess. We just do `image.src` and `image.alt`. See the [usual oe_theme](https://github.com/openeuropa/oe_theme/blob/0877a9568616af27518ffa3beb34514520fc60ae/templates/patterns/text_featured_media/pattern-text-featured-media.html.twig#L48).
In card we did different, because the card BCL template needs markup. So in the pattern preprocess we make the transformation from the expected pattern variable (again an object) to the final render array.

**Additional notes**:
We could, in the future, ditch even the transformation in render array done in the preprocess, as we can invoke directly the method in twig. The thing is that at this moment I added the `toRenderArray()`  method only to our implementation of ImageValueObjectInterface, so we need to do some conversion. This is ideally temporary, and we should add the method to the interface.